esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -1845,16 +1845,10 @@ void CFileSprayEx::readAndCheckSpraySourceReq(MemoryBuffer& srcxml, const char*
                 Owned<IPropertyTree> match = getHostGroup(hostGroup,true);
                 if (!isEmptyString(srcIP))
                 {
-                    Owned<IPropertyTreeIterator> hostIter = match->getElements("hosts");
-                    ForEach (*hostIter)
-                    {
-                        const char *knownIP = hostIter->query().queryProp(nullptr);
-                        if (strcmp(knownIP, srcIP)==0)
-                        {
-                            sourceIPReq.set(knownIP);
-                            return;
-                        }
-                    }
+                    VStringBuffer xexpr(".[hosts='%s']", srcIP);
+                    if (!match->hasProp(xexpr))
+                        throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "srcip %s is not valid storage plane %s", srcIP, srcPlane);
+                    sourceIPReq.set(match->queryProp(srcIP));
```

Comment:
I think you mean

    sourceIPReq.set(srcIP);

Reply:
@ghalliday @shamser - looks like this still needs to be fixed?

Reply:
sorry, looks like I was commend on an out of date commit. Ignore above comment.